code/__HELPERS/text.dm

```diff
@@ -469,3 +469,38 @@ var/list/rot13_lookup = list()
 			t_out += char
 
 	return t_out
+
+/proc/dmm_encode(text)
+	// First, go through and nix out any of our escape sequences so we don't leave ourselves open to some escape sequence attack
+	// Some coder will probably despise me for this, years down the line
+
+	var/list/repl_chars = list("&#34;", "&#123;", "&#125;")
+	for(var/char in repl_chars)
+		var/index = findtext(text, char)
+		var/keylength = length(char)
+		while(index)
+			log_debug("Bad string given to dmm encoder! [text]")
+			text = copytext(text, 1, index) + copytext(text, index+keylength)
+			index = findtext(text, char)
+
+	// Then, replace characters as normal
+	var/list/repl_chars_2 = list("\"" = "&#34;", "{" = "&#123;", "}" = "&#125;")

```

Comment:
& isn't a special character in the dmm parser, but it IS a special character here. This parsing scheme would probably benefit from escaping it, though. Presently this code can be tricked by the following text:

`&#3&#123;4;`
This will get past the code above to strip out the escape sequences. If I replaced those with another character, like `_`, then that form of attack would be plugged, however.